[PyTorch] remove string_view::operator[] bounds check

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #64824
* #64823
* #64691
* #64839
* #64838
* #64822
* #64821
* #64820
* #64784
* __->__ #64670
* #64682
* #64678

Bounds checking is not required for `std::string_view`, and the checking hoses performance for the following performance prototype diff.

Differential Revision: [D30747515](https://our.internmc.facebook.com/intern/diff/D30747515/)

swolchok committed:
[PyTorch] remove string_view bounds check

Bounds checking is not required for `std::string_view`, and the checking hoses performance for the following performance prototype diff.

Differential Revision: [D30747515](https://our.internmc.facebook.com/intern/diff/D30747515/)

[ghstack-poisoned]

bhosmer reviewed (approved):
Agreed that a polyfill for `std::string_view` should conform to the standard definition for it. (The discrepancy isn't mentioned in the comment header, so I'm assuming it was unintentional)

swolchok committed:
Update on "[PyTorch] remove string_view::operator[] bounds check"


Bounds checking is not required for `std::string_view`, and the checking hoses performance for the following performance prototype diff.

Differential Revision: [D30747515](https://our.internmc.facebook.com/intern/diff/D30747515/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] remove string_view::operator[] bounds check"


Bounds checking is not required for `std::string_view`, and the checking hoses performance for the following performance prototype diff.

Differential Revision: [D30747515](https://our.internmc.facebook.com/intern/diff/D30747515/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] remove string_view::operator[] bounds check"


Bounds checking is not required for `std::string_view`, and the checking hoses performance for the following performance prototype diff.

Differential Revision: [D30747515](https://our.internmc.facebook.com/intern/diff/D30747515/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] remove string_view::operator[] bounds check"


Bounds checking is not required for `std::string_view`, and the checking hoses performance for the following performance prototype diff.

Differential Revision: [D30747515](https://our.internmc.facebook.com/intern/diff/D30747515/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] remove string_view::operator[] bounds check"


Bounds checking is not required for `std::string_view`, and the checking hoses performance for the following performance prototype diff.

Differential Revision: [D30747515](https://our.internmc.facebook.com/intern/diff/D30747515/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] remove string_view::operator[] bounds check"


Bounds checking is not required for `std::string_view`, and the checking hoses performance for the following performance prototype diff.

Differential Revision: [D30747515](https://our.internmc.facebook.com/intern/diff/D30747515/)

[ghstack-poisoned]
